src/main/java/com/google/sps/data/GivrUser.java

```diff
@@ -51,48 +53,118 @@ public boolean isMaintainer() {
     return this.isMaintainer;
   }
 
-  public static GivrUser getUserByIdFromDatastore(String userId) {
+  public String getUserEmail() {
+    return this.email;
+  }
+
+  public boolean isLoggedIn() {
+    return this.isLoggedIn;
+  }
+
+  public static PreparedQuery getPreparedQueryResultUserWithProperty(String propertyName, String propertyValue) {
     DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    Filter queryFilter = new FilterPredicate("userId", FilterOperator.EQUAL, userId);
+
+    Filter queryFilter = null;
+    if (propertyName.equals("isMaintainer")) {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, Boolean.parseBoolean(propertyValue));
+    } else {
+      queryFilter = new FilterPredicate(propertyName, FilterOperator.EQUAL, propertyValue);
+    }
     Query query = new Query("User").setFilter(queryFilter);
-      
-    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+
     PreparedQuery preparedQuery = datastore.prepare(query);
+    return preparedQuery;
+  }
+
+  // Check if User with propertyName, propertyValue exists within Datastore.
+  public static boolean checkIfUserWithPropertyExists(String propertyName, String propertyValue) {
+    PreparedQuery preparedQuery = getPreparedQueryResultUserWithProperty(propertyName, propertyValue);
+    FetchOptions fetchOptions = FetchOptions.Builder.withLimit(1);
+    QueryResultList<Entity> userResult = preparedQuery.asQueryResultList(fetchOptions);
+
+    if (userResult.size() < 1) {
+      return false;
+    } else if (userResult.size() > 1) {
+      throw new IllegalArgumentException("More than one user with the " + propertyName + " was found.");
+    }
+    return true;
+  }
+
+  // Updates a User entity in Datastore, identifying with the first two parameters with values from second two parameters.
+  public static void updateUserInDatastore(String identifyingProperty, String identifyingValue, String updateProperty, String updateValue) {  
```

Comment:
What other parameters would you expect for identifyingProperty? Why limit to a single update property?

Reply:
Does this mean to have a method that can take a variable number of properties?

So, something like updateUserInDatastore(String identifyingProperty, String identifyingValue, Map<String, String> propertiesAndValuesToUpdate)?

Reply:
Yup.

Any reason to make these statics?

Why not:

GivrUser user = GivrUser.GetUserFromProperty("userId", foobar);
user.setEmail("email");
user.commitInTransaction(); 

?

Reply:
I made this function static because the entity that's being changed would depend on the identifying property, not the object that this function is called on (if this function is not static).